picard/config.py

```diff
@@ -234,10 +242,25 @@ def run_upgrade_hooks(self, outputfunc=None):
             self._version = PICARD_VERSION
             self._write_version()
 
+    def _backup_settings(self):
+        if Version(0, 0, 0) < self._version < PICARD_VERSION:
+            backup_path = self._versioned_config_filename()
+            log.info('Backing up config file to %s', backup_path)
+            try:
+                shutil.copyfile(self.fileName(), backup_path)
+            except OSError:
+                log.error('Failed backing up config file to %s', backup_path)
```

Comment:
@phw : I'd also catch and ignore other exceptions, to ensure this never prevents Picard to start.
History of `shutil.copyfile()` shows evolution of raised exceptions, and we can't be sure new ones will not appear in future Python versions.
What do you think?

Reply:
Let's keep it more explicit for now. I will submit a separate PR to handle unexpected exceptions better and provide better user feedback for those cases.